### PR TITLE
Add PMP function is_polygon_soup_a_polygon_mesh

### PR DESCRIPTION
**Pull request by bo0ts**
https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Small_Features/PMP_is_polygon_soup_a_polygon_mesh

Add a function to check if a polygon soup is a polygon mesh

### CODE REVIEW

**Comment by janetournois** [comment]
I have updated this PR and pushed some improvements discussed in the small feature submission in the branch cgal-dev/Pmp-poly_soup_poly_mesh-pmoeller

@bo0ts, @lrineau or @sloriot can you please push the updated branch to @bo0ts's fork?

**Comment by lrineau** [comment]
> @bo0ts, @lrineau or @sloriot can you please push the updated branch to @bo0ts's fork?

Done. I had to push with `--force`: you have rebased the branch onto `master`.

**Comment by janetournois** [comment]
I did it on purpose, thanks

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/polygon_soup_to_polygon_mesh.h [hunk 1]
```diff
@@ -91,45 +96,78 @@ class Polygon_soup_to_polygon_mesh
...
+          return false;//vertex met twice in the same polygon
+      }
     }
 
-    return true;
+    //check manifoldness
+    std::sort(points.begin(), points.end());
+    std::unique(points.begin(), points.end());
+
+    std::vector<Polygon> polygons2(polygons.begin(), polygons.end());
```
Comment (sloriot) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/polygon_soup_to_polygon_mesh.h]: I have a patch locally to remove the need of filling points. I need to think how we could avoid copying polygons.

**Comment by sloriot** [comment]
I've updated the branch on cgal-dev to remove  `points` and `polygons2`. Now we need to update the static functions so that the requirements on PolygonRange are met by those functions..

**Comment by janetournois** [comment]
I've done the modifications necessary to make doc and code consistent.

I also removed the hard-coded `std::vector`'s from the `PMP::orient_polygon_soup` API

**Comment by sloriot** [comment]
@lrineau could you please push cgal-dev branch into @bo0ts' fork?

**Comment by lrineau** [comment]
done

**Comment by janetournois** [comment]
there will be more changes in the documentation before this PR gets merged. The small feature is still under discussion

**Comment by sloriot** [comment]
Just updated the branch.

**Comment by sloriot** [comment]
See [this error](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.9-Ic-35/Polygon_mesh_processing/TestReport_lrineau_Ubuntu-Latest-NO_DEPRECATED_CODE.gz)

**Comment by janetournois** [comment]
I'm taking care of it

**Comment by sloriot** [comment]
Wiki updated

**Pull request merged**